### PR TITLE
Add STANAG ranger plates

### PR DESCRIPTION
**Pull request by Ilurker**
#### Summary
CONTENT "Adds ranger plates to some STANAG box magazines"

#### Purpose of change

Implements "ranger plate" type attachments commonly used with STANAG-type magazines. Ranger plates are small carabiner attachments affixed to the base plates of compact magazines to allow them to clip onto various attachments.

#### Describe the solution

Spawn rates have been adjusted to represent a small share of 5, 10, 20, 30 and 40 round magazines having these attachments in place, with a somewhat higher share being encountered in gun shops, alongside less common rifle types, and among the civilian population. Spawn rates for the corresponding STANAG-type magazines have been slightly adjusted downwards in situations where ranger plates might spawn to keep the total availability of magazines more or less constant with current balance.

Ranger plates do not spawn alongside standard issue firearms like the M4 or the HK416, as it's less likely rank and file soldiers would be providing their own equipment.

#### Describe alternatives you've considered

Expanding the gun modification system to allow modifications to magazines as well.

#### Testing

Spawned items in question and verified they attach appropriately and did not result in any errors.

#### Additional context

https://magpul.com/pmagrangerplate-genm35-56x45-3pack.html?mp_global_color=118

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
The link is dead?

**Comment by GuardianDll** [comment]
Link works fine for me
Im not really sure, what this plates is used for?

**Comment by Ilurker** [comment]
They're multi-function, but one use is to use the loop to attach a magazine to a belt clip or carabiner clip without using a pouch. 

The alternatives are to re-describe them to be more in line with this:

https://www.cryeprecision.com/ProductDetail/accmc100000_magclip-set-of-3

Essentially allowing players the option to utilize a belt clip slot to hold a single magazine (so long as it's a variant with the clip installed).

The ranger plate style magazine itself could technically also provide slightly faster reload and better stability (as they're designed to assist with both), but I thought the effect would be too subtle to justify the code required.

File: data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json [hunk 1]
```diff
@@ -1340,7 +1340,8 @@
     "entries": [
       { "item": "fs2000", "charges-min": 0, "charges-max": 30 },
       { "item": "stanag30" },
```
Comment (bombasticSlacks) [inline data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json:1342]:
```suggestion
      { "group": "stanag_30_random" },
```

File: data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json [hunk 2]
```diff
@@ -1340,7 +1340,8 @@
...
-      { "item": "stanag30", "prob": 50 },
+      { "item": "stanag30ranger", "prob": 10 },
```
Comment (bombasticSlacks) [inline data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json:1343]:
```suggestion
      { "group": "stanag_30_random", "prob": 50 },
```

File: data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json [hunk 3]
```diff
@@ -1340,7 +1340,8 @@
...
+      { "item": "stanag30", "prob": 40 },
```
Comment (bombasticSlacks) [inline data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json:1344]:
```suggestion

```

File: data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json [hunk 4]
```diff
@@ -1513,7 +1514,8 @@
     "entries": [
       { "item": "iwi_tavor_x95_300blk", "charges-min": 0, "charges-max": 30 },
       { "item": "stanag30" },
```
Comment (bombasticSlacks) [inline data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json:1516]:
```suggestion
      { "group": "stanag_30_random" },
```

File: data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json [hunk 5]
```diff
@@ -1513,7 +1514,8 @@
...
-      { "item": "stanag30", "prob": 50 },
+      { "item": "stanag30ranger", "prob": 10 },
```
Comment (bombasticSlacks) [inline data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json:1517]:
```suggestion
      { "group": "stanag_30_random", "prob": 50 },
```

File: data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json [hunk 6]
```diff
@@ -1513,7 +1514,8 @@
...
+      { "item": "stanag30", "prob": 40 },
```
Comment (bombasticSlacks) [inline data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json:1518]:
```suggestion
```

File: data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json [hunk 7]
```diff
@@ -1549,7 +1551,8 @@
     "entries": [
       { "item": "m27_assault_rifle", "variant": "scar_l", "charges-min": 0, "charges-max": 30 },
       { "item": "stanag30" },
```
Comment (bombasticSlacks) [inline data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json:1553]:
```suggestion
      { "group": "stanag_30_random" },
```

File: data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json [hunk 8]
```diff
@@ -1549,7 +1551,8 @@
...
-      { "item": "stanag30", "prob": 50 },
+      { "item": "stanag30ranger", "prob": 10 },
```
Comment (bombasticSlacks) [inline data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json:1554]:
```suggestion
      { "group": "stanag_30_random", "prob": 50 },
```

File: data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json [hunk 9]
```diff
@@ -1549,7 +1551,8 @@
...
+      { "item": "stanag30", "prob": 40 },
```
Comment (bombasticSlacks) [inline data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json:1555]:
```suggestion

```

File: data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json [hunk 10]
```diff
@@ -1804,7 +1807,8 @@
     "entries": [
       { "item": "oa93", "charges-min": 0, "charges-max": 30 },
       { "item": "stanag10" },
```
Comment (bombasticSlacks) [inline data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json:1809]:
```suggestion
      { "group": "stanag_10_random" },
```

File: data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json [hunk 11]
```diff
@@ -1804,7 +1807,8 @@
...
-      { "item": "stanag10", "prob": 50 },
+      { "item": "stanag10ranger", "prob": 25 },
```
Comment (bombasticSlacks) [inline data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json:1810]:
```suggestion
      { "group": "stanag_10_random", "prob": 50 },
```

File: data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json [hunk 12]
```diff
@@ -1804,7 +1807,8 @@
...
+      { "item": "stanag10", "prob": 33 },
```
Comment (bombasticSlacks) [inline data/json/itemgroups/Weapons_Mods_Ammo/nested_guns.json:1811]:
```suggestion
```

**Review by bombasticSlacks — COMMENTED** [review]
Overall this looks great, I'm glad to see these added it's a cool addition. 

However for the changes to the civvie guns spawn groups they are supposed to have 1 mag for sure and possibly a second at 50% across the board so adding your ranger mags as a % chance can lead to up to three mags. There isn't for any reason other than consistency, so people know if they use any of those groups they get very consistent results. Also it's a shame that you can't end up with two mags with ranger plates with a gun. 

you can resolve this with the suggestions I've made + adding these item groups to magazines.json

```json
  {
    "id": "stanag_30_random",
    "type": "item_group",
    "//": "one of the possible 30 round stanags",
    "subtype": "distribution",
    "entries": [
      { "item": "stanag30", "prob": 80 },
      { "item": "stanag30ranger", "prob": 20 }
    ]
  },
  {
    "id": "stanag_10_random",
    "type": "item_group",
    "//": "one of the possible 10 round stanags",
    "subtype": "distribution",
    "entries": [
      { "item": "stanag10", "prob": 80 },
      { "item": "stanag10ranger", "prob": 20 }
    ]
  }
```

**Comment by Ilurker** [comment]
Thank you for the suggestion! That is actually what I was aiming for, but I didn't know the best way to do it - it's been pushed.

File: data/json/items/magazine/223.json [hunk 13]
```diff
@@ -64,6 +64,24 @@
     "flags": [ "MAG_COMPACT" ],
     "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "223": 10, "300blk": 10 } } ]
   },
+  {
+    "looks_like": "stanag30",
+    "id": "stanag10ranger",
+    "type": "MAGAZINE",
+    "name": { "str": "STANAG 10-round magazine with attached ranger plate" },
```
Comment (dseguin) [inline data/json/items/magazine/223.json:71]:
```suggestion
    "name": {
      "str": "STANAG 10-round magazine with attached ranger plate",
      "str_pl": "STANAG 10-round magazines with attached ranger plates"
    },
```

File: data/json/items/magazine/223.json [hunk 14]
```diff
@@ -64,6 +64,24 @@
...
+    "description": "A compact 10-round box magazine for use with STANAG compatible rifles. The bottom has been modified with a ranger plate attachment to allow it to clip to carabiners.",
```
Comment (dseguin) [inline data/json/items/magazine/223.json:72]:
```suggestion
    "description": "A compact 10-round box magazine for use with STANAG compatible rifles.  The bottom has been modified with a ranger plate attachment to allow it to clip to carabiners.",
```

File: data/json/items/magazine/223.json [hunk 15]
```diff
@@ -138,6 +156,23 @@
     "flags": [ "MAG_COMPACT" ],
     "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "223": 5, "300blk": 5 } } ]
   },
+  {
+    "id": "stanag5ranger",
+    "looks_like": "stanag30",
+    "type": "MAGAZINE",
+    "name": { "str": "STANAG 5-round magazine with ranger plate" },
```
Comment (dseguin) [inline data/json/items/magazine/223.json:163]:
```suggestion
    "name": { "str": "STANAG 5-round magazine with ranger plate", "str_pl": "STANAG 5-round magazines with ranger plates" },
```

File: data/json/items/magazine/223.json [hunk 16]
```diff
@@ -138,6 +156,23 @@
...
+    "description": "A very compact 5-round box magazine for use with STANAG compatible rifles. The bottom has been modified with a ranger plate attachment to allow it to clip to carabiners.",
```
Comment (dseguin) [inline data/json/items/magazine/223.json:164]:
```suggestion
    "description": "A very compact 5-round box magazine for use with STANAG compatible rifles.  The bottom has been modified with a ranger plate attachment to allow it to clip to carabiners.",
```

File: data/json/items/magazine/223.json [hunk 17]
```diff
@@ -155,6 +190,23 @@
     "flags": [ "MAG_COMPACT" ],
     "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "223": 20, "300blk": 20 } } ]
   },
+  {
+    "id": "stanag20ranger",
+    "looks_like": "stanag30",
+    "type": "MAGAZINE",
+    "name": { "str": "STANAG 20-round magazine with ranger plate" },
```
Comment (dseguin) [inline data/json/items/magazine/223.json:197]:
```suggestion
    "name": { "str": "STANAG 20-round magazine with ranger plate", "str_pl": "STANAG 20-round magazines with ranger plates" },
```

File: data/json/items/magazine/223.json [hunk 18]
```diff
@@ -155,6 +193,23 @@
     "flags": [ "MAG_COMPACT" ],
     "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "223": 20, "300blk": 20 } } ]
   },
+  {
+    "id": "stanag20ranger",
+    "looks_like": "stanag30",
+    "type": "MAGAZINE",
+    "name": { "str": "STANAG 20-round magazine with ranger plate", "str_pl": "STANAG 20-round magazines with ranger plates" },
+    "description": "A compact 20-round box magazine for use with STANAG compatible rifles. The bottom has been modified with a ranger plate attachment to allow it to clip to carabiners.",
```
Comment (dseguin) [inline data/json/items/magazine/223.json:201]:
```suggestion
    "description": "A compact 20-round box magazine for use with STANAG compatible rifles.  The bottom has been modified with a ranger plate attachment to allow it to clip to carabiners.",
```

File: data/json/items/magazine/223.json [hunk 19]
```diff
@@ -173,6 +228,24 @@
     "flags": [ "MAG_COMPACT" ],
     "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "223": 30, "300blk": 30 } } ]
   },
+  {
+    "id": "stanag30ranger",
+    "type": "MAGAZINE",
+    "name": { "str": "STANAG 30-round magazine with ranger plate" },
+    "//": "Tileset whitelist for rifle magazines",
+    "description": "A standard capacity 30-round box magazine for use with STANAG compatible rifles. The bottom has been modified with a ranger plate attachment to allow it to clip to carabiners.",
```
Comment (dseguin) [inline data/json/items/magazine/223.json:236]:
```suggestion
    "description": "A standard capacity 30-round box magazine for use with STANAG compatible rifles.  The bottom has been modified with a ranger plate attachment to allow it to clip to carabiners.",
```

File: data/json/items/magazine/223.json [hunk 20]
```diff
@@ -190,6 +263,23 @@
     "flags": [ "MAG_COMPACT" ],
     "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "223": 40, "300blk": 40 } } ]
   },
+  {
+    "id": "stanag40ranger",
+    "looks_like": "stanag30",
+    "type": "MAGAZINE",
+    "name": { "str": "STANAG 40-round magazine with ranger plate" },
```
Comment (dseguin) [inline data/json/items/magazine/223.json:270]:
```suggestion
    "name": { "str": "STANAG 40-round magazine with ranger plate", "str_pl": "STANAG 40-round magazines with ranger plates" },
```

File: data/json/items/magazine/223.json [hunk 21]
```diff
@@ -190,6 +263,23 @@
...
+    "description": "An increased-capacity 40-round box magazine for use with STANAG compatible rifles. The bottom has been modified with a ranger plate attachment to allow it to clip to carabiners.",
```
Comment (dseguin) [inline data/json/items/magazine/223.json:271]:
```suggestion
    "description": "An increased-capacity 40-round box magazine for use with STANAG compatible rifles.  The bottom has been modified with a ranger plate attachment to allow it to clip to carabiners.",
```

**Pull request merged**